src/providers/azure/mod.rs

```diff
@@ -418,12 +418,28 @@ impl Azure {
             .chain_err(|| "failed to get hostname")?;
         Ok(name)
     }
+
+    fn fetch_sku (&self) -> Result<Option<String>> {
```

Comment:
Why is there an inner `Option<T>` here? Can the SKU be missing/empty?
I think we can either return a plain `Result<T>` (if the SKU should always be present) or an `Option<T>` (if the SKU can be missing/empty).

Reply:
The logic I took is the same as `fetch_host` on line# 407.
From my understanding, the return type is coming from https://github.com/coreos/afterburn/blob/59d3c56ccefe093ec98d5aecb2579b88b7b5e9b5/src/retry/client.rs#L195 which also depends on https://github.com/coreos/afterburn/blob/59d3c56ccefe093ec98d5aecb2579b88b7b5e9b5/src/retry/client.rs#L238 the request would fail so the outer `Result<_>` and if response is 404 then return `Ok<None>` hence the inner `Option<String>`.. Am I missing something?

Reply:
No, you are not missing anything. But the client API you are quoting is a generic HTTP client with retries, which also supports pages with empty bodies (thus the `Result<Option<T>>` return type).
In this specific case, I think you can map the result to a simpler `Result<String>`.

Reply:
I see, thanks for explaining!